rust/docs/parsing-specification.md

```
@@ -0,0 +1,165 @@
+Operators are, from highest to lowest precedence (and always left-associative):
+
+- postfix repetition, e.g. `a*` (0 or more), `a+` (1 or more), `a?` (0 or 1),
+	`a{n}` (exactly `n`), and `a{min,max}` (at least `min`, up to and including `max` times)
+- binary concatenation, e.g. `a*b`, equivalent to `(a*)b`
+- binary alternation ("or"), e.g. `a*b|c`, equivalent to `((a*)b)|c`
+
+Furthermore, a root rule's pattern may be anchored with a leading `^` or ending `$`,
+meaning the match must be preceded/followed by one of the delimiter characters
+(or the very start or end of input).
+
+More explicitly, pattern syntax follows this [EBNF][ebnf] grammar.
```

markdownlint-cli2 (0.22.1)

[warning] Reference links and images should use a label that is defined
Missing link or image reference definition: "ebnf"

(MD052, reference-links-images)